src/bin/nydusd/main.rs

```diff
@@ -276,6 +276,21 @@ fn prepare_commandline_options() -> Command {
                 .required(false)
                 .global(true),
         )
+        .arg(
+        Arg::new("systemsock")
+            .long("systemsock")
+            .help("Path to the Nydus-Snapshotter system API socket")
+            .required(false)
+            .global(true),
+        )
+        .arg(
+        Arg::new("credential-source")
```

Comment:
How about renaming it to `backend-source`, and combining the url with `systemsock`?